src/Transpile.ts

```diff
@@ -27,14 +27,20 @@ export interface TranspileOptions {
     customTransformers?: ts.CustomTransformers;
     transformer?: LuaTransformer;
     printer?: LuaPrinter;
+    emitHost?: EmitHost;
+}
+
+export interface EmitHost {
+    readFile: (path: string, encoding?: string | undefined) => string | undefined;
```

Comment:
If we never specify encoding we can just drop this argument to simplify custom implementations (ts does that with few hosts). Also, it could be a bit more consistent with method syntax:

```suggestion
    readFile(path: string): string | undefined;
```